docs/source/contribution/technical_steering_committee.md

```diff
@@ -1,13 +1,16 @@
 # Join the Technical Steering Committee
 
-The term "Technical Steering Committee" (TSC) describes the group of Kedro maintainers.
+In the Kedro project's latest iteration it is an incubating project within [LF AI & Data](https://lfaidata.foundation/).
```

Comment:
Somehow this sentence feels a bit difficult to read but I don't have a better idea 😓